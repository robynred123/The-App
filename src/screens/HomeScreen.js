import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

const HomeScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Text>Welcome!</Text>
      <Text>Please select a screen to view</Text>
      <View>
        <TouchableOpacity
          style={styles.navigationButton}
          onPress={() => navigation.navigate('About')}
        >
          <Text>About Me</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.navigationButton}
          onPress={() => navigation.navigate('Developer')}
        >
          <Text>Why Me? Why Asda?</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  navigationButton: {
    backgroundColor: '#b067d8',
    alignContent: 'space-between'
  }
});

export default HomeScreen