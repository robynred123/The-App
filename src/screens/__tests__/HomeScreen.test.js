import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

import HomeScreen from '../HomeScreen';

describe('<HomeScreen />', () => {
    it('should render the HomeScreen without crashing', () => {
        const component = shallow(<HomeScreen />)
        expect(component).toMatchSnapshot()
    });

    it('should render the HomeScreen with text', () => {
        const component = shallow(<HomeScreen />)
        expect(component.find('Text').at(0).children().text()).toEqual('Welcome!')
        expect(component).toMatchSnapshot()
    });

    it('should render the HomeScreen with an "about me" button', () => {
        const component = shallow(<HomeScreen />)
        const button = component.find('MenuButton').at(0).props().find('title')
        expect(button).toEqual('About Me')
        
    });

    // it('should render the HomeScreen with an "about me" button', () => {
    //     const component = shallow(<HomeScreen />)
    //     const button = component.find('MenuButton').at(1).props()
    //     expect(button).toEqual('{"navigateTo: "Developer", "navigation": undefined, "title": "Why Me? Why Asda?"}')
    // });
})