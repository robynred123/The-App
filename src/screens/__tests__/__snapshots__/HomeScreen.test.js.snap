// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render the HomeScreen with text 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 80,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
      Welcome!
    </Text>
    <Text>
      Please select a screen to view
    </Text>
  </View>
  <View>
    <MenuButton
      navigateTo="About"
      title="About Me"
    />
    <MenuButton
      navigateTo="Developer"
      title="Why Me? Why Asda?"
    />
  </View>
</View>
`;

exports[`<HomeScreen /> should render the HomeScreen without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 80,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
      Welcome!
    </Text>
    <Text>
      Please select a screen to view
    </Text>
  </View>
  <View>
    <MenuButton
      navigateTo="About"
      title="About Me"
    />
    <MenuButton
      navigateTo="Developer"
      title="Why Me? Why Asda?"
    />
  </View>
</View>
`;
